generals/agents.py

```diff
@@ -28,13 +30,15 @@ def play(self, observation):
         """
         Randomly selects a valid action.
         """
-        mask = observation['action_mask']
+        mask = observation["action_mask"]
         valid_actions = np.argwhere(mask == 1)
         action_index = np.random.choice(len(valid_actions))
-        # append 1 or 0 randomly to the action (to say whether to send half of troops or all troops)
-        action = np.append(valid_actions[action_index], np.random.choice([0, 1]))
+
+        # [1] to indicate we want to move, [0] to indicate we want to move all troops
+        action = np.concatenate(([1], valid_actions[action_index], [0]))
```

Comment:
I liked that `RandomClient` can sometimes move only half of troops, but maybe half of the time is really too much.

Reply:
will fix as discussed.